toolkit/tools/pkg/imagecustomizerlib/liveosisobuilder.go

```diff
@@ -242,31 +250,99 @@ func (b *LiveOSIsoBuilder) prepareRootfsForDracut(writeableRootfsDir string) err
 	return nil
 }
 
-func mergeConfigs(savedConfigsFilePath string, newKernelArgs imagecustomizerapi.KernelExtraArguments) (mergedConfigs *SavedConfigs, err error) {
+// mergeConfigs
+//
+//		This function merges:
+//	 - a subset of the user current input configuration yaml.
+//	 - a subset of the user saved (from previous runs) input configuration yaml.
+//
+//	 Depending on the configuration the behavior can be:
+//	 - Concatenation:
+//	   - like in the case of iso-specific kernel parameters.
+//	 - Replacement:
+//	   - like in the case of the pxeImageUrl
+//
+//	 In case of replacement, priority is given to the new configuration if
+//	 present.
+//
+// inputs:
+//   - savedConfigsFilePath:
+//     full path to the yaml configuration file hold configuration from previous
+//     runs.
+//   - newKernelArgs:
+//     kernel argument specified by the user in this run.
+//   - newPxeIsoImageUrl:
+//     PXE ISO image URL specified by the user in this run.
+//   - newOSDracutVersion:
+//     Dracut package version of the rootfs provided by the user.
+//
+// outputs:
+// - returns a SavedConfigs objects with the new merged values.
+func mergeConfigs(savedConfigsFilePath string, newKernelArgs imagecustomizerapi.KernelExtraArguments,
```

Comment:
This function doesn't just merge the configs. It also writes the merged config to disk. Perhaps change the name?

(It might also be goodness to have the merge logic in a separate function that only handles the merge logic This might make the code more self-documenting.)

Reply:
renamed.
Left them together though. I considered splitting it per suggestions, but I guess the new name `updateSavedConfigs` is a bit more expressive.